File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityHeadLookPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record EntityHeadLookPacket(int entityId, byte headYaw) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(headYaw);
    }

    @Override
    public byte getId() {
        return 0x19;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityHeadLookPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityHeadLookPacket(val entityId: Int, val headYaw: Byte) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(headYaw.toInt())
    }

    override val id: Byte
        get() = 0x19
}
